tiledb/sm/consolidator/group_meta_consolidator.cc

```diff
@@ -113,11 +113,12 @@ void GroupMetaConsolidator::vacuum(const char* group_name) {
 
   // Get the group metadata URIs and vacuum file URIs to be vacuumed
   auto& vfs = resources_.vfs();
-  auto compute_tp = storage_manager_->compute_tp();
-  GroupDirectory group_dir;
+  auto& compute_tp = resources_.compute_tp();
+  shared_ptr<GroupDirectory> group_dir;
```

Comment:
Whenever you might feel the urge to use an allocated variable to solve a scope problem, always ask if the scope problem is fundamental. In this case the `try` block in the next statement is the problem, not something to try to preserve.

* The `throw` statement in the `catch` block is throwing a `Status` instance, which is Just Wrong. It should be throwing `GroupDirectoryException`, but that class doesn't exist yet. The `Status_` function should be replace with an exception class.
* The constructor of `GroupDirectory` is throwing `logic_error` for I/O problems. It's the wrong exception for it to be throwing. It should be throwing `GroupDirectoryException`. See previous point.
* Once the constructor is throwing the right exception, there's no value at all in the `try` block. It can be omitted.